Pomodoro.Api/Controllers/TasksController.cs

```diff
@@ -0,0 +1,190 @@
+            data.UserId = new Guid("6FF9D0D2-BB26-468C-E841-08DB10C9B2BE");
+
+            /*if (this.UserId == Guid.Empty)
+            {
+                this.Unauthorized();
+            }*/
+
+            var result = await this.tasksService.PostTask(data);
+            return this.CreatedAtAction(nameof(this.GetTaskById), new { id = result.TaskId }, result);
+        }
+
+        /// <summary>
+        /// Endpoint to delete a task.
+        /// </summary>
+        /// <param name="task">Gets as a parameter Task view model.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [HttpDelete("deleteTask")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(200, "The task was created")]
+        [SwaggerResponse(401, "An unauthorized request cannot be processed.")]
+        [SwaggerResponse(404, "The task you want to delete wan`t found.")]
+        [SwaggerResponse(500, "Something went wrong")]
+        public async Task<ActionResult> DeleteTask(TaskToManipulateModel task)
+        {
+            var data = this.mapper.Map<TaskToManipulateModel, TaskModel>(task);
+            data.UserId = new Guid("6FF9D0D2-BB26-468C-E841-08DB10C9B2BE");
+
+            try
+            {
+                await this.tasksService.DeleteTask(data);
+            }
+            catch (Exception ex)
+            {
+                return this.NotFound(ex);
+            }
+
+            return this.Ok();
+        }
+
+        /// <summary>
+        /// Endpoint to update a task.
+        /// </summary>
+        /// <param name="task">Gets as a parameter Task view model.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [HttpPut("updateTask")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(200, "The task was created")]
+        [SwaggerResponse(401, "An unauthorized request cannot be processed.")]
+        [SwaggerResponse(404, "The task you want to update wan`t found.")]
+        [SwaggerResponse(500, "Something went wrong")]
+        public async Task<ActionResult> UpdateTask(TaskToManipulateModel task)
+        {
+            var data = this.mapper.Map<TaskToManipulateModel, TaskModel>(task);
+            data.UserId = new Guid("6FF9D0D2-BB26-468C-E841-08DB10C9B2BE");
+            await this.tasksService.UpdateTask(data);
+            return this.Ok();
+        }
+    }
+}
```

Comment:
Need to rewrite: add authorization, don't check empty Guid, don't hardcode userId, etc.